docs/swarm-workflow/swarm-workflow-state-interaction-duality/README.md

```diff
@@ -0,0 +1,84 @@
+# Interaction-State Duality with Machine-Runner
+
+This document expands the idea of how a set of complex interaction can be represented as a state when formalized with Machine Runner's swarm protocol.
+If you haven't read the parent document, please do so [here](../) because this document will use the same example-problem.
+
+## The Problem-Solution as A Sequence
+
+From the problem presented in the [parent document](../) we end up with the following solution, a sequence of local tasks and interactions.
+
+```text
+1. robot waits & pump ensures the dock is clear (Initial)
+2. pump signals: "dock available"
+3. robot docks & pump waits
+4. robot signals: "docking successful"
+5. robot waits & pump supplies water
+6. pump signals: water supplied
+7. robot undocks & pump waits
+8. robot signals: undock successful
+9. (Done)
+```
+
+This solution assumes that we are using Machine Runner's swarm protocol so that agents can easily communicate with the entire swarm;
+the word signals in the line `pump signals: "dock available` means that an agent (in this case `pump`) sends an event (in this case `dock available`), which is then received by the entire swarm.
+
+### Sequence Diagram
+
+Crudely and directly translated into this sequence diagram.
```

Comment:
You could complete the sentence by adding a subject, but I think it can just be removed.